datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -238,6 +238,16 @@ impl CommonSubexprEliminate {
         let rewritten = pop_expr(&mut rewritten)?;
 
         if affected_id.is_empty() {
+            // Alias aggregation expressions if they have changed
+            // TODO: This should be handled in `build_recover_project_plan` once qualified aliases
```

Comment:
I think this comment is out of date

Reply:
Yup, removed and merged latest main.

Thanks for your patience on this one! :)

Reply:
> Thanks for your patience on this one! :)

It is **I** who should be thanking **you**.

Thanks again @gruuya